lib/adf_diag.py

```diff
@@ -778,185 +810,273 @@ def create_plots(self):
-
-###############
+                # PRECT can be found by simply adding PRECL and PRECC
+                # grab file names for the PRECL and PRECC files from the case ts directory
+                if glob.glob(os.path.join(ts_dir, "*PRECC*")) and glob.glob(
+                    os.path.join(ts_dir, "*PRECL*")
+                ):
+                    constit_files = sorted(glob.glob(os.path.join(ts_dir, "*PREC*")))
+                else:
+                    ermsg = "PRECC and PRECL were not both present; PRECT cannot be calculated."
+                    ermsg += " Please remove PRECT from diag_var_list or find the relevant CAM files."
+                    raise FileNotFoundError(ermsg)
+                # create new file name for PRECT
+                prect_file = constit_files[0].replace("PRECC", "PRECT")
+                if Path(prect_file).is_file():
+                    if overwrite:
+                        Path(prect_file).unlink()
+                    else:
+                        print(
+                            f"[{__name__}] Warning: PRECT file was found and overwrite is False. Will use existing file."
+                        )
+                        return None
+                # append PRECC to the file containing PRECL
+                os.system(f"ncks -A -v PRECC {constit_files[0]} {constit_files[1]}")
+                # create new file with the sum of PRECC and PRECL
+                os.system(
+                    f"ncap2 -s 'PRECT=(PRECC+PRECL)' {constit_files[1]} {prect_file}"
+                )
+            if var == "RESTOM":
+                # RESTOM = FSNT-FLNT
+                # Have to be more precise than with PRECT because FSNTOA, FSTNC, etc are valid variables
+                if glob.glob(os.path.join(ts_dir, "*.FSNT.*")) and glob.glob(
+                    os.path.join(ts_dir, "*.FLNT.*")
+                ):
+                    input_files = [
+                        sorted(glob.glob(os.path.join(ts_dir, f"*.{v}.*")))
+                        for v in ["FLNT", "FSNT"]
+                    ]
+                    constit_files = []
+                    for elem in input_files:
+                        constit_files += elem
+                else:
+                    ermsg = "FSNT and FLNT were not both present; RESTOM cannot be calculated."
+                    ermsg += " Please remove RESTOM from diag_var_list or find the relevant CAM files."
+                    raise FileNotFoundError(ermsg)
+                # create new file name for RESTOM
+                derived_file = constit_files[0].replace("FLNT", "RESTOM")
+                if Path(derived_file).is_file():
+                    if overwrite:
+                        Path(derived_file).unlink()
+                    else:
+                        print(
+                            f"[{__name__}] Warning: RESTOM file was found and overwrite is False. Will use existing file."
+                        )
+                        return None
```

Comment:
I think you want to use `continue` here instead:

```suggestion
                        continue
```

Reply:
Done.